### PR TITLE
Add equals() and hashcode() methods to endpoint classes

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Methods are required to use EndPoints as a key in hash maps

@stephenc @reviewbybees

### CODE REVIEW

**Comment by rsandell** [comment]
:+1:

File: src/main/java/org/jenkinsci/plugins/docker/commons/DockerRegistryEndpoint.java [hunk 1]
```diff
@@ -159,6 +159,32 @@ public String imageName(String userAndRepo) throws IOException {
...
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        final DockerRegistryEndpoint other = (DockerRegistryEndpoint) obj;
+        if ((this.url == null) ? (other.url != null) : !this.url.equals(other.url)) {
+            return false;
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/docker/commons/DockerRegistryEndpoint.java]: Can be simplified using `ObjectUtils.equals` (Commons Lang) or `Objects.equal` (Guava).

Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/docker/commons/DockerRegistryEndpoint.java]: e.g.

```java
@Override public boolean equals(Object obj) {
    return obj instanceof DockerRegistryEndpoint &&
           ObjectUtils.equals(url, ((DockerRegistryEndpoint) obj).url) &&
           ObjectUtils.equals(credentialsId, ((DockerRegistryEndpoint) obj).credentialsId);
}
```

**Comment by jglick** [comment]
:+1:

File: src/main/java/org/jenkinsci/plugins/docker/commons/DockerRegistryEndpoint.java (hunk 1 above)
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/docker/commons/DockerRegistryEndpoint.java]: @jglick 
Sorry, I've missed the comment. <code>ObjectUtils</code> is deprecated by <code>java.util.Objects</code>, but it does not change the comment's idea to reuse the existing code. This idea is definitely good. Unfortunately, all methods in classes fall-back to <code>A.equals(B)</code>.

There could be a generic reflection-based Lib method, but a manual implementation is faster, which is significant for the class usage in containers. I would propose to leave the change as is.

Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/docker/commons/DockerRegistryEndpoint.java]: > `ObjectUtils` is deprecated by `java.util.Objects`

Yes, once we are on Java 7+.

This was not important, just a thought.

**Pull request merged**